scripts/fiss_form_QA_prep_lucy.R

```diff
@@ -0,0 +1,56 @@
+# script to add data from habitat_confirmations.xlsm to form_fiss_2023.gpkg because edits were made in habitat_confirmations.xlsm
+# Moving forward we should do a QA of form_fiss_2023.gpkg in Q then populate the habitat_confirmations.xlsm.
+
+source('scripts/packages.R')
+
+dir_project <- 'sern_skeena_2023'
+
+
+## Import the raw form_fiss_2023.gpkg
+form_fiss_site_raw <- fpr::fpr_sp_gpkg_backup(
+  path_gpkg = paste0("~/Projects/gis/", dir_project, '/data_field/2023/form_fiss_site_2023.gpkg'),
+  update_utm = TRUE,
+  update_site_id = FALSE,
+  write_back_to_path = FALSE,
+  write_to_csv = FALSE,
+  write_to_rdata = FALSE,
+  return_object = TRUE)
+
+
+## Import the habitat_confirmations.xlsm
+habitat_confirmations <- fpr::fpr_import_hab_con(row_empty_remove = T)
+
+## Weird workflow here because hand edits were done to temp, cond, and turb data in step 2 therefore it has the most up
+## to date temp, cond, and turb data (this data is also in step 4 but its not up to date). Temp, cond, and turbidity
+## needs to be extracted from step 2 and added to form_fiss_site_2023.gpkg.
+
+## Extract temp, cond, turb data from step_2_fish_coll_data
+fish_coll_data <- habitat_confirmations %>%
+  purrr::pluck("step_2_fish_coll_data") %>%
+  select(local_name, temperature_c, conductivity_m_s_cm, turbidity)
+
+## Add temp, cond, and turb data from fish_coll_data to form_fiss_site_2023.gpkg
+form_fiss_prep1 <- dplyr::left_join(form_fiss_site_raw %>%
+                               #remove target columns that are in form_fiss_site_raw so we can add them from fish_coll_data
+                               select(-c(temperature_c, conductivity_m_s_cm, turbidity)),
+                              fish_coll_data %>%
+                               distinct(local_name, temperature_c, conductivity_m_s_cm, turbidity),
+                              by = join_by("local_name")) %>%
```

Comment:
`join_by` is not necessary.  it is more appropriately used to reference how to join complicated tibble pieces outside of the join functions.  see ?join_by

can just be `by = "local_name"`

Reply:
okay yup. I know we can just use `by` but when I was going through R4DS they suggested using `join_by` because it "provides a clearer and more flexible specification." according to them, see section here https://r4ds.hadley.nz/joins#mutating-joins.

Can remove though.